--- minecraft/src/net/minecraft/entity/item/EntityFallingBlock.java
+++ src/main/java/net/minecraft/entity/item/EntityFallingBlock.java
@@ -20,10 +20,11 @@
 import net.minecraft.network.datasync.EntityDataManager;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.world.World;
 
 public class EntityFallingBlock extends Entity {
@@ -70,10 +71,20 @@
 
    public boolean canBeCollidedWith() {
       return !this.isDead;
    }
 
+   // Paper Start
+   public boolean isOnGround() {
+       BlockPos blockBeneath = new BlockPos(posX, posY - 0.009999999776482582D, posZ);
+       if(!BlockFalling.canFallThrough(worldObj.getBlockState(blockBeneath))) return true;
+       if(BlockFalling.canFallThrough(worldObj.getBlockState(blockBeneath.down()))) return false;
+       ArrayList<AxisAlignedBB> list = new ArrayList<AxisAlignedBB>();
+       this.worldObj.getBlockState(blockBeneath.down()).addCollisionBoxToList(worldObj, blockBeneath, getCollisionBoundingBox(), list, this);
+       return list.size() > 0;
+   }
+   // Paper End
    public void onUpdate() {
       Block var1 = this.fallTile.getBlock();
       if(this.fallTile.getMaterial() == Material.AIR) {
          this.setDead();
       } else {
@@ -97,11 +108,11 @@
          this.motionZ *= 0.9800000190734863D;
          if(!this.worldObj.isRemote) {
             BlockPos var9 = new BlockPos(this);
             if(this.onGround) {
                IBlockState var3 = this.worldObj.getBlockState(var9);
-               if(BlockFalling.canFallThrough(this.worldObj.getBlockState(new BlockPos(this.posX, this.posY - 0.009999999776482582D, this.posZ)))) {
+               if(!isOnGround()) {
                   this.onGround = false;
                   return;
                }
 
                this.motionX *= 0.699999988079071D;
@@ -151,11 +162,11 @@
    public void fall(float var1, float var2) {
       Block var3 = this.fallTile.getBlock();
       if(this.hurtEntities) {
          int var4 = MathHelper.ceiling_float_int(var1 - 1.0F);
          if(var4 > 0) {
-            ArrayList var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
+            ArrayList<Entity> var5 = Lists.newArrayList(this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.getEntityBoundingBox()));
             boolean var6 = var3 == Blocks.ANVIL;
             DamageSource var7 = var6?DamageSource.anvil:DamageSource.fallingBlock;
 
             for(Entity var9 : var5) {
                var9.attackEntityFrom(var7, (float)Math.min(MathHelper.floor_float((float)var4 * this.fallHurtAmount), this.fallHurtMax));
